internal/timerange/timerange.go

```diff
@@ -0,0 +1,23 @@
+package timerange
+
+import (
+	"errors"
+	"time"
+)
+
+type Range struct {
+	Start *time.Time
+	End   *time.Time
+}
+
+func New(start, end *time.Time) (Range, error) {
+	// The end time must be greater than or equal to the start time.
+	if end != nil && start != nil && end.Before(*start) {
+		return Range{}, errors.New("new time range: end time cannot be before start time")
+	}
+	return Range{Start: start, End: end}, nil
+}
+
+func NewInstant(t *time.Time) Range {
```

Comment:
// NewInstant returns a Range where the Start and End times are both set to the given time.